src/data/real/irrational.lean

```diff
@@ -257,4 +257,52 @@ open irrational
 @[simp] theorem irrational_inv_iff : irrational x⁻¹ ↔ irrational x :=
 ⟨of_inv, irrational.inv⟩
 
+/-!
+### Lemmas about Irrationals between Reals
+-/
+
+/-- An irrational number between any two distinct rational numbers.-/
+noncomputable def irrational_btwn_rats {x y : ℚ} (h : x < y) : { r : ℝ // irrational r } :=
+⟨x +((y - x) : ℚ) * (real.sqrt 2)⁻¹, begin
+  refine irrational.rat_add _
+    (irrational.rat_mul
+      (irrational_inv_iff.mpr irrational_sqrt_two) _),
+  exact_mod_cast (ne_of_gt (sub_pos_of_lt h)) end⟩
+
+lemma lt_irrational_btwn_rats {x y : ℚ} (h : x < y) : (x : ℝ) < (irrational_btwn_rats h : ℝ) :=
+begin
+  suffices : 0 < ↑(y - x) * (real.sqrt 2)⁻¹,
+  { simpa [irrational_btwn_rats] using h, },
+  refine (zero_lt_mul_right _).mpr _,
+  apply inv_pos.mpr,
+  norm_num,
+  exact_mod_cast (sub_pos_of_lt h)
+end
+
+lemma irrational_btwn_rats_lt {x y : ℚ} (h : x < y) : (irrational_btwn_rats h : ℝ) < y :=
```

Comment:
You have multiple goals midway through this proof, I think. You need to delimit the proof sections handling the separate goals using braces.